tests/unit/amo/components/TestAutoSearchInput.js

```diff
@@ -133,6 +133,14 @@ describe(__filename, () => {
       expect(root).toHaveState('searchValue', '');
     });
 
+    it('sets the search value to an empty string when `location.query` is not a string', () => {
+      const location = createFakeLocation({ query: { query: ['a', 'b'] } });
```

Comment:
can we test with queries in the form of array indexes too, like `q[0]`?

Reply:
That's what I am attempting to do here. It's `react-router` that is taking the value from the URL and populating `location.query`, so when we are manually creating a `location` here in the test, we are providing a value that looks like what `react-router` would have done. It converts a querystring like `?query[0]=a&query[1]=b` into an array that looks like `query: ['a', 'b']`, so that's what I am using in the test.

Really, because the new code excludes anything that isn't a string, we could have used any non-string value in this test, but I wanted to use something that resembles what is happening in the real world with this bug.

Does that make sense?

Reply:
I wouldn't have guessed that `query: ['a', 'b']` would be converted to `?query[0]=a&query[1]=b` (I would have guessed something like `?query=a&query=b` or `?query=[a,b]`) but sure?

Reply:
To satisfy my curiosity could we have an assertion on the following line that `createFakeLocation` _does_ render `location` to a string like that?

Reply:
I don't think we can do exactly what you want. `createFakeLocation` generates an object that looks like a `location` that has already been parsed by `react-router`, so nothing in there is going to resemble `?query[0]=a&query[1]=b`. That's what the URL looks like *before* it is parsed, but we are eliminating the parsing code and just directly creating the result.

Reply:
🤷 I guess we have to trust that it's what would be generated then (the problem with unit tests)

Reply:
Yes, but, as I mentioned above, the test is perfectly good for testing the code that was changed, as all it's doing it looking for anything that isn't a string. We really need a functional test to test that the bug is fixed, but that's why I always test things manually as well.